watertap/unit_models/electrolyzer.py

```diff
@@ -0,0 +1,731 @@
+        )
+
+        # ---------------------------------------------------------------------
+        # performance
+
+        @self.Constraint(
+            doc="voltage efficiency as a function of minimum required by reaction"
+        )
+        def eq_efficiency_voltage(b):
+            return b.voltage_min == b.voltage_applied * b.efficiency_voltage
+
+        @self.Constraint(
+            doc="electrons passed between anode and cathode contributing to reactions"
+        )
+        def eq_electron_flow(b):
+            return (
+                b.electron_flow
+                * pyunits.convert(
+                    Constants.faraday_constant,
+                    to_units=units_meta("time")
+                    * units_meta("current")
+                    / units_meta("amount"),
+                )
+                == b.current * b.efficiency_current
+            )
+
+        @self.Constraint(doc="membrane area")
+        def eq_membrane_area(b):
+            return b.current_density * b.membrane_area == b.current
+
+        # TODO: find relationship between required surface area of electrode as a function of process (flowrate/volume)
+        #  or allow for custom area term without universal current density term
+        # assumed
+        @self.Constraint(doc="anode area")
+        def eq_anode_area(b):
+            return b.current_density * b.anode_area == b.current
+
+        # assumed
+        @self.Constraint(doc="cathode area")
+        def eq_cathode_area(b):
+            return b.current_density * b.cathode_area == b.current
+
+        @self.Constraint(doc="energy consumption")
+        def eq_power(b):
+            return b.power == b.voltage_applied * b.current
+
+        @self.Constraint(doc="energy efficiency")
+        def eq_efficiency_power(b):
+            return b.efficiency_power == b.efficiency_current * b.efficiency_voltage
+
+        # ---------------------------------------------------------------------
+        # reactions
+
+        # fix stoichiometry to zero to be overwritten when constructing flowsheet
+        for p in self.config.property_package.phase_list:
+            for j in self.config.property_package.component_list:
+                self.anode_stoich[p, j].fix(0)
+                self.cathode_stoich[p, j].fix(0)
+                self.membrane_stoich[p, j].fix(0)
+
+        # ---------------------------------------------------------------------
+        # mass balance
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="ion permeation through the membrane",
+        )
+        def eq_membrane_permeation(b, t, p, j):
+            return (
+                b.anolyte.mass_transfer_term[t, p, j]
+                == -b.electron_flow * b.membrane_stoich[p, j]
+            )
+
```

Comment:
I am trying to line up the codes with the current documentation.  could you write out the full mass/material balance eq in the doc, as N_in - N_out + N_r = 0 ?  If you have balance for both ion and molecule species, I think the stoich chef. should appear in the balance eq ?

Reply:
I agree that it wasn't exactly intuitive so I added the following to the documentation.
![image](https://user-images.githubusercontent.com/101219154/234598430-e02a886a-9b9e-4f8d-8e55-73f6564f63a8.png)